### PR TITLE
20_2 TreeList Adaptability: rewrite

### PR DESCRIPTION
**Pull request by pavel-2020**
[Trello](https://trello.com/c/9Xi72KLo/1656-treelist-demos-adaptability-1-demo)
Based on https://js.devexpress.com/Demos/WidgetsGallery/Demo/DataGrid/GridAdaptabilityOverview/jQuery/Light/
https://github.com/DevExpress/devextreme-demos/pull/828 https://github.com/DevExpress/devextreme-demos/pull/829

### CODE REVIEW

File: JSDemos/Demos/TreeList/Adaptability/description.md [hunk 1]
```diff
@@ -1 +1,5 @@
...
\ No newline at end of file
+TreeList automatically adapts its layout to screens with different sizes. In this demo, you can switch between horizontal and vertical screen orientations. If columns do not fit the selected orientation, TreeList hides them one by one, starting with the rightmost column. Information from the hidden columns is still available in adaptive detail rows.
+
+To enable this feature, set the [columnHidingEnabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#columnHidingEnabled) property to **true**.
+
+You can also use the **columns[]**.[hidingPriority](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#hidingPriority) property to specify custom hiding priorities for those columns that you want to hide. Other columns will never be hidden. The values are zero-based; columns with lower values are hidden first.
```
Comment (RomanTsukanov) [inline JSDemos/Demos/TreeList/Adaptability/description.md]:
```suggestion
The TreeList hides columns based on their hiding priorities. Columns with lower priority values are hidden first. You can use the **columns[]**.[hidingPriority](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#hidingPriority) property to specify custom hiding priorities for those columns that you want to hide. Other columns will never be hidden.
```

**Review by RomanTsukanov — CHANGES_REQUESTED** [review]

**Review by RomanTsukanov — DISMISSED** [review]

File: JSDemos/Demos/TreeList/Adaptability/description.md [hunk 2]
```diff
@@ -1 +1,5 @@
-The TreeList can adapt its layout to&nbsp;narrow screens. If&nbsp;the screen width is&nbsp;not enough to&nbsp;display all columns, the component hides them according to&nbsp;the specified priority. To&nbsp;view the values of&nbsp;hidden columns, click the ellipsis button in&nbsp;the required row. Switch between the landscape and portrait layouts to&nbsp;see how the TreeList looks on&nbsp;wide or&nbsp;narrow screens.
\ No newline at end of file
+TreeList automatically adapts its layout to screens with different sizes. In this demo, you can switch between horizontal and vertical screen orientations. If columns do not fit the selected orientation, TreeList hides them one by one, starting with the rightmost column. Information from the hidden columns is still available in adaptive detail rows.
```
Comment (arminal) [inline JSDemos/Demos/TreeList/Adaptability/description.md]:
```suggestion
The TreeList automatically adapts its layout to screens with different sizes. In this demo, you can switch between horizontal and vertical screen orientations. If columns do not fit the selected orientation, the TreeList hides them one by one, starting with the rightmost column. Information from the hidden columns is still available in adaptive detail rows.
```

File: JSDemos/Demos/TreeList/Adaptability/description.md [hunk 3]
```diff
@@ -1 +1,5 @@
...
+
+To enable this feature, set the [columnHidingEnabled](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/#columnHidingEnabled) property to **true**.
+
+The TreeList hides columns based on their hiding priorities. Columns with lower priority values are hidden first. You can use the **columns[]**.[hidingPriority](/Documentation/ApiReference/UI_Components/dxTreeList/Configuration/columns/#hidingPriority) property to specify custom hiding priorities for those columns that you want to hide. Other columns will never be hidden.
```
Comment (arminal) [inline JSDemos/Demos/TreeList/Adaptability/description.md:5]: hiding priorities?

Comment (pavel-2020) [reply JSDemos/Demos/TreeList/Adaptability/description.md:5]: Yes, they determine the order that TreeList uses to hide columns.

Comment (arminal) [reply JSDemos/Demos/TreeList/Adaptability/description.md:5]: ok, that's fine

**Review by RomanTsukanov — APPROVED** [review]

**Pull request merged**